contracts/SponsorFaucet.sol

```diff
@@ -64,38 +109,180 @@ contract SponsorFaucet is Ownable, Pausable, ReentrancyGuard {
     function isAppliable(address dapp) public returns (bool) {
         if (_isAppliableForGas(dapp) || _isAppliableForCollateral(dapp))
             return true;
+        require(dapp.isContract(), "ERROR_ADDRESS_IS_NOT_CONTRACT");
         _validateApplyForGas(dapp);
         _validateApplyForCollateral(dapp);
     }
 
+    //accept sponsor's cfx
+    function() external payable {}
+
+    //withdraw to specific address by amount
+    function withdraw(address payable sponsor, uint256 amount)
+        public
+        onlyOwner
+        nonReentrant
+        whenPaused
```

Comment:
How about to remove this Pause management? From user's point of view, `Paused` and `Balance not enough` are both failures, and we can simplify the code here.

Reply:
rm this from **withdraw**, apply function can still be paused/unpaused.